src/System.IO.FileSystem/src/System/IO/Directory.cs

```diff
@@ -227,17 +227,11 @@ public static string GetDirectoryRoot(string path)
                 throw new ArgumentNullException(nameof(path));
 
             string fullPath = Path.GetFullPath(path);
-            string root = fullPath.Substring(0, PathInternal.GetRootLength(fullPath.AsSpan()));
+            string root = Path.GetPathRoot(fullPath);
```

Comment:
@JeremyKuhne found this 3rd instance where GetPathRoot could be called.

Reply:
This one I wouldn't change as this does add measurable overhead to a simple API.